src/app/application/components/data-source/DataSource.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+
+import { watch, ref, onBeforeUnmount } from 'vue'
+
+import { useDataSource } from '@/utilities'
+
+const data = useDataSource()
+
+const props = defineProps({
+  src: {
+    type: String,
+    required: true,
+  },
+})
+
+const message = ref<unknown | undefined>(undefined)
```

Comment:
**Suggestion**: This shouldn’t include the type `undefined` as `unknown | undefined` is just `unknown`.

```suggestion
const message = ref<unknown>(undefined)
```

Reply:
Ah I think this was when I was messing around with generic components, I've taken the `undefined` out for now.